replication-api-impl/src/test/java/org/codice/ditto/replication/api/impl/MetacardConfigLoaderTest.java

```diff
@@ -0,0 +1,338 @@
+import ddf.catalog.data.impl.MetacardTypeImpl;
+import ddf.catalog.data.impl.types.CoreAttributes;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.source.IngestException;
+import java.io.Serializable;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import org.codice.ddf.configuration.SystemBaseUrl;
+import org.codice.ditto.replication.api.Direction;
+import org.codice.ditto.replication.api.ReplicationException;
+import org.codice.ditto.replication.api.ReplicationType;
+import org.codice.ditto.replication.api.ReplicatorConfig;
+import org.codice.ditto.replication.api.impl.data.ReplicatorConfigImpl;
+import org.codice.ditto.replication.api.impl.mcard.ReplicationConfigAttributes;
+import org.codice.ditto.replication.api.impl.modern.ReplicationSiteImpl;
+import org.codice.ditto.replication.api.mcard.ReplicationConfig;
+import org.codice.ditto.replication.api.modern.ReplicationSite;
+import org.codice.ditto.replication.api.modern.ReplicationSitePersistentStore;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.opengis.filter.Filter;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MetacardConfigLoaderTest {
+
+  MetacardConfigLoader configLoader;
+
+  @Mock CatalogFramework framework;
+
+  @Mock MetacardHelper helper;
+
+  @Mock ReplicationSitePersistentStore siteStore;
+
+  MetacardType type;
+
+  @Before
+  public void setUp() throws Exception {
+    FilterBuilder builder = new GeotoolsFilterBuilder();
+    List<MetacardType> types = new ArrayList<>();
+    types.add(new CoreAttributes());
+    types.add(new ReplicationConfigAttributes());
+    type = new MetacardTypeImpl("replication-config", types);
+    doCallRealMethod()
+        .when(helper)
+        .setIfPresent(any(Metacard.class), any(String.class), any(Serializable.class));
+    doCallRealMethod()
+        .when(helper)
+        .setIfPresentOrDefault(
+            any(Metacard.class),
+            any(String.class),
+            any(Serializable.class),
+            any(Serializable.class));
+    when(helper.getAttributeValueOrDefault(any(Metacard.class), anyString(), any(Object.class)))
+        .thenCallRealMethod();
+    configLoader = new MetacardConfigLoader(framework, builder, helper, type, siteStore);
+    System.setProperty(SystemBaseUrl.INTERNAL_HOST, "localhost");
+    System.setProperty(SystemBaseUrl.INTERNAL_HTTPS_PORT, "5678");
+  }
+
+  @AfterClass
+  public static void afterClass() {
```

Comment:
✏️ We can alternatively use the `RestoreSystemProperties` rule from codice-test
https://github.com/codice/codice-test/blob/master/junit-extensions/src/main/java/org/codice/junit/RestoreSystemProperties.java

Reply:
👍 Much better alternative

Reply:
Nice. Changed.